contracts/external/lib/Bytes.sol

```diff
@@ -0,0 +1,196 @@
+/*
+
+    Copyright 2019 dYdX Trading Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+*/
+
+pragma solidity 0.5.7;
+pragma experimental ABIEncoderV2;
+
+import { Require } from "../../protocol/lib/Require.sol";
+
+
+/**
+ * @title Bytes
+ * @author dYdX
+ *
+ * Library to manipulate bytes
+ */
+library Bytes {
+    // ============ Constants ============
+
+    bytes32 constant FILE = "Bytes";
+
+    // ============ Functions ============
+
+    /**
+     * Gets the memory address for the contents of a byte array.
+     *
+     * @param  input  Byte array to lookup.
+     * @return        Memory address of the contents of the byte array.
+     */
+    /* solium-disable-next-line security/no-named-returns */
+    function contentAddress(bytes memory input)
+        internal
+        pure
+        returns (uint256 memoryAddress)
+    {
+        /* solium-disable-next-line security/no-inline-assembly */
+        assembly {
+            memoryAddress := add(input, 32)
+        }
+        return memoryAddress;
+    }
+
+    /**
+     * Writes a bytes32 into a specific position in a byte array.
+     *
+     * @param  b      Byte array to insert <input> into.
+     * @param  index  Index in byte array of <input>.
+     * @param  input  bytes32 to put into byte array.
+     */
+    function writeBytes32(
+        bytes memory b,
+        uint256 index,
+        bytes32 input
+    )
+        internal
+        pure
+    {
+        Require.that(
+            b.length >= index + 32,
+            FILE,
+            "Cannot write 32 bytes"
+        );
+
+        /* solium-disable-next-line security/no-inline-assembly */
+        assembly {
+            mstore(add(add(b, 32), index), input)
+        }
+    }
+
+    /**
+     * Copies `length` bytes from memory location `source` to `dest`.
+     *
+     * @param  dest    memory address to copy bytes to.
+     * @param  source  memory address to copy bytes from.
+     * @param  length  number of bytes to copy.
+     */
+    function memCopy(
```

Comment:
this function is really dangerous, complicated, and scary...

Reply:
we only use it 3 places, I wonder if it's worth it

Reply:
removed function